src/index.js

```diff
@@ -25,13 +22,24 @@ class PubsubBaseProtocol extends EventEmitter {
   /**
    * @param {String} debugName
    * @param {String} multicodec
-   * @param {Object} libp2p libp2p implementation
-   * @param {Object} options
-   * @param {boolean} options.signMessages if messages should be signed, defaults to true
-   * @param {boolean} options.strictSigning if message signing should be required, defaults to true
+   * @param {PeerInfo} peerInfo
+   * @param {Object} registrar
+   * @param {function} registrar.register
+   * @param {function} registrar.unregister
+   * @param {Object} [options]
+   * @param {boolean} [options.signMessages] if messages should be signed, defaults to true
+   * @param {boolean} [options.strictSigning] if message signing should be required, defaults to true
```

Comment:
My idea here is to add the topology to the options, which would overwrite the ones defined by the protocol. Should we do it here, or iterate the registrar topology in a following PR?  My main concern is to add docs to stuff that we will not support yet